src/main/kotlin/study/racing/domain/Cars.kt

```diff
@@ -0,0 +1,54 @@
+package study.racing.domain
+
+import study.racing.strategy.MoveStrategy
+import study.racing.strategy.RandomMoveStrategy
+
+class Cars private constructor(
+    private var cars: List<Car> = listOf(),
+    private val strategy: MoveStrategy = RandomMoveStrategy()
+) {
+
+    init {
+        require(cars.isNotEmpty()) {
+            "차량은 최소 1대 이상 요청해 주세요."
+        }
+    }
+
+    fun moveTheCars() {
+        cars.forEach { car ->
+            car.tryMoveTheCar(strategy.isMoving())
+        }
+    }
+
+    fun getCarsMoveDistance(): List<Int> = this.cars.map {
+        it.getCarDistance()
+    }.toList()
+
+    companion object {
+        fun from(
+            carCount: Int
+        ): Cars {
+            return Cars(
+                createCountOfCars(carCount)
+            )
+        }
+
+        fun of(
+            carCount: Int,
+            strategy: MoveStrategy
+        ): Cars {
+            return Cars(
+                createCountOfCars(carCount),
+                strategy
+            )
+        }
+
+        private fun createCountOfCars(carCount: Int): List<Car> {
+            val cars = mutableListOf<Car>()
+            repeat(carCount) {
+                cars.add(Car())
+            }
+            return cars
+        }
```

Comment:
```suggestion
private fun createCountOfCars(carCount: Int): List<Car> = List(carCount) { Car() }
```

이렇게도 간략하게 만들 수 있답니다. 요번 기회에 코틀린의 Collection 함수에 대해서 공부해보는 건 어떨까요? 생각보다 많은 기능들이 있답니다.